lib/DiscoveryStreamFeed.jsm

```diff
@@ -455,6 +457,23 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     this._sendSpocsFill({...filtered, frequency_cap: frequencyCapped}, true);
   }
 
+  async clearSpocs() {
+    const endpoint = this.store.getState().Prefs.values[PREF_SPOCS_CLEAR_ENDPOINT];
+    if (!endpoint) {
+      return;
+    }
+    const headers = new Headers();
+    headers.append("content-type", "application/json");
+
+    await this.fetchFromEndpoint(endpoint, {
```

Comment:
To be clear, `fetchFromEndpoint` goes through the allowed endpoints check which I think is desired but affects what endpoints we use or might need to update that pref at the same time as setting the clear endpoint

Reply:
The endpoint to clear the data is intended to be on the same endpoint that gets spocs, so it's, for example "https://getpocket.cdn.mozilla.net/v3/firefox/unique-spocs" with a POST and "https://getpocket.cdn.mozilla.net/v3/firefox/user" with a DELETE request. So whatever the endpoint ends up being, we'll need to update it in a patch anyway, and it'll match an existing endpoint so we probably won't need to update the allowed endpoints any more than we already need to, which is once at most.